### PR TITLE
Upgrade blocks

### PR DESCRIPTION
**Pull request by par12005**
# New Feature

Issue #18 

## Description

 - Upgraded implementation of Powered By Tripal, Notifications, and Content Type Bar Chart blocks.
 - Added Tripal Logo images to `tripal/images` folder.
 - The `admin/dashboard` page no longer exists by default in Drupal, so I added it to `tripal/tripal.routing.yml`
 - Added `tripal/css/tripal.css`, which is nearly the same as Tripal 3's [`tripal/theme/css/tripal.css`](https://github.com/tripal/tripal/blob/7.x-3.x/tripal/theme/css/tripal.css)
 - Separated D3 JS library from Tripal JS library and updated local D3 JS (`tripal/js/d3.min.js` -> `tripal/js/d3.v3.min.js`).
 - Added Tripal Dashboard JS library to `tripal/tripal.libraries.yml`.
 - Added `tripal_bundle` table.

## Testing

 - Re-install Tripal 4 in order to create the `tripal_bundle` table.
 - Clear your cache
 - Add some data to the `tripal_bundle` table and a `chado_bio_data_N` table, for example:
```SQL
INSERT INTO tripal_bundle (type, term_id, name, label) VALUES
    ('TripalEntity', 2, 'bio_data_2', 'Analysis'),
    ('TripalEntity', 3, 'bio_data_3', 'Organism');
CREATE SEQUENCE chado_bio_data_2_mapping_id_seq START 1;
CREATE SEQUENCE chado_bio_data_3_mapping_id_seq START 1;
CREATE TABLE chado_bio_data_2 (
    mapping_id INTEGER NOT NULL DEFAULT nextval('chado_bio_data_2_mapping_id_seq'::regclass) 
    PRIMARY KEY,
    entity_id INTEGER NOT NULL,
    record_id INTEGER NOT NULL,
    nid INTEGER);
CREATE TABLE chado_bio_data_3 (
    mapping_id INTEGER NOT NULL DEFAULT nextval('chado_bio_data_3_mapping_id_seq'::regclass) 
    PRIMARY KEY,
    entity_id INTEGER NOT NULL,
    record_id INTEGER NOT NULL,
    nid INTEGER);
INSERT INTO chado_bio_data_2 (mapping_id, entity_id, record_id) VALUES
    (1, 38496, 7),
    (2, 26551, 5),
    (3, 4786, 2);
GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA public TO drupal;
```
 - Add some data to the `tripal_admin_notfications` table, for example:
```SQL
INSERT INTO tripal_admin_notfications (details, title, actions, submitter_id, type) VALUES
('some details', 'title', 'a:1:{s:15:"Go to dashboard";s:16:"tripal.dashboard";}', 1, 'Field');
```
 - Navigate to `admin/dashboard`
 - Verify that the notifications and content type bar chart blocks both work as expected.
 - Navigate to `admin/structure/block`
 - Choose a block layout area and click "Place block"
 - In the menu, scroll down to the "Powered by Tripal" block and click "Place block"
 - Choose your logo size and type, and your visibility restrictions, then click "Save block"
 - Navigate to a page which satisfies your visibility restrictions you set in the previous step, and verify that the Powered by Tripal block appears on the page.

## Additional Notes

I am waiting to mark this PR as "ready to merge" because I would like to have the `tripal_admin_notfications` table spelled correctly (PR #42) before this code is merged, since the Notifications block depends on that table.

Also, the "Dismiss notification" links in the Notifications block currently do nothing, so I'm wondering if those links count as part of the block upgrade issue (#18). If so, I can implement that and add it to this PR.

### CODE REVIEW

**Comment by par12005** [comment]
The Notifications block is now using the `tripal_admin_notifications` table, so this PR should be ready for review.

**Comment by laceysanderson** [comment]
Hi @par12005, thanks for the contribution! 

In Tripal 4 the `tripal_bundle` table changed to `tripal_entity` to follow Drupal 8 conventions. You can see this table in action by creating Tripal Content Types at Structure > Tripal Content Types. We don't yet have chado integration for these entities but I intend to follow the same `chado_bio_data_N` structure. As such, we don't want this PR to create a tripal_bundle table. Also note, the tripal_entity table uses the id to form the `bio_data` name rather then having a name column which could become out of sync. The label becomes the title.

Regarding the "dismiss notifications" we can either make that a new issue or you can add it to this PR -whichever you prefer :-)

Otherwise, this looks good :-) We'll likely upgrade the dashboard page to a panel in the future but this works for now!

**Comment by par12005** [comment]
Thanks @laceysanderson! I've just pushed 2 more commits which remove the tripal_bundle table and use the tripal_entity table to get the counts of different tripal entities instead. If you could take a look when you get the chance that would be great!

File: tripal/js/tripal.dashboard.js [hunk 1]
```diff
@@ -75,7 +75,11 @@
       }
 
       // Now insert the bar chart.
-      barchart2('#tripal-entity-type-chart', '#block-tripal-content-type-barchart', entityCountListing);
+      console.log('test');
```
Comment (laceysanderson) [inline tripal/js/tripal.dashboard.js]: Remove debugging statement :-)
```suggestion
```

File: tripal/src/Plugin/Block/NotificationsBlock.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+ * )
+ */
+class NotificationsBlock extends BlockBase implements BlockPluginInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $db = \Drupal::database();
+    $output = "hello";
```
Comment (laceysanderson) [inline tripal/src/Plugin/Block/NotificationsBlock.php]: This variable is never used and contains placeholder text.
```suggestion
```

**Review by laceysanderson — APPROVED** [review]
Functional Review
✅ Powered by Tripal block
    - love different sizes and colour options 👍 
✅ Tripal Content Barchart
    - Great that it is now chado agnostic 👍 
    - Responds well to different magnitudes of content
    - could use a message when there is no content 🤷‍♀️ 
    - I like that it only shows content types that there is content for (different from T3)
✅ Tripal Notifications
     - degrades well if there is no tripal_notifications table
     - handles lack of messages well
     - shows messages :-)
✅ Tripal Dashboard
     - works 👍 
     - Could use a menu link in the admin menu

As mentioned, the Dismiss button does not work yet. I appreciate that it provides a message to that effect and no errors :-) Do you want to open an issue @par12005 indicating that the dismiss button still needs to be upgraded so we don't forget?

I added a couple of suggestions (remove debugging and placeholder code) but once those are addressed I think this PR is ready to be merged 👍

**Comment by laceysanderson** [comment]
Thank you!

**Pull request merged**